k8s/results/environment.txt
Sun Oct  4 04:12:24 PM UTC 2026
Client Version: v1.36.2-eks-8f14419
Kustomize Version: v5.8.1
Server Version: v1.34.11-eks-cfb47f5
Warning: version difference between client (1.36) and server (1.34) exceeds the supported minor version skew of +/-1
NAME                                            STATUS   ROLES    AGE   VERSION                INTERNAL-IP      EXTERNAL-IP      OS-IMAGE                        KERNEL-VERSION                     CONTAINER-RUNTIME
ip-192-168-33-205.eu-north-1.compute.internal   Ready    <none>   10h   v1.34.11-eks-3b4a6ca   192.168.33.205   16.192.166.166   Amazon Linux 2023.12.20260928   6.12.110-135.201.amzn2023.x86_64   containerd://2.2.7+unknown
ip-192-168-86-95.eu-north-1.compute.internal    Ready    <none>   10h   v1.34.11-eks-3b4a6ca   192.168.86.95    16.171.193.158   Amazon Linux 2023.12.20260928   6.12.110-135.201.amzn2023.x86_64   containerd://2.2.7+unknown
ip-192-168-9-242.eu-north-1.compute.internal    Ready    <none>   10h   v1.34.11-eks-3b4a6ca   192.168.9.242    13.49.18.189     Amazon Linux 2023.12.20260928   6.12.110-135.201.amzn2023.x86_64   containerd://2.2.7+unknown
NAME                       READY   UP-TO-DATE   AVAILABLE   AGE     CONTAINERS   IMAGES                                                      SELECTOR
deployment.apps/postgres   1/1     1            1           9h      postgres     postgres:12                                                 app=postgres
deployment.apps/redis      1/1     1            1           9h      redis        redis:7                                                     app=redis
deployment.apps/web        1/1     1            1           7h52m   web          <ACCOUNT_ID>.dkr.ecr.eu-north-1.amazonaws.com/jobqueue:v1   app=web
deployment.apps/worker     8/8     8            8           7h52m   worker       <ACCOUNT_ID>.dkr.ecr.eu-north-1.amazonaws.com/jobqueue:v1   app=worker

NAME               TYPE           CLUSTER-IP       EXTERNAL-IP                                                                PORT(S)        AGE     SELECTOR
service/postgres   ClusterIP      10.100.3.95      <none>                                                                     5432/TCP       9h      app=postgres
service/redis      ClusterIP      10.100.125.135   <none>                                                                     6379/TCP       9h      app=redis
service/web        LoadBalancer   10.100.254.107   a64fef4ee63204f2d997f5e9f1c1d521-1749294773.eu-north-1.elb.amazonaws.com   80:32448/TCP   7h52m   app=web

NAME                           READY   STATUS      RESTARTS   AGE     IP               NODE                                            NOMINATED NODE   READINESS GATES
pod/jobqueue-migrate-fxbxt     0/1     Completed   0          8h      192.168.46.226   ip-192-168-33-205.eu-north-1.compute.internal   <none>           <none>
pod/monitor                    1/1     Running     0          4h13m   192.168.46.226   ip-192-168-33-205.eu-north-1.compute.internal   <none>           <none>
pod/postgres-cf9fdbc59-tfwlk   1/1     Running     0          9h      192.168.85.124   ip-192-168-86-95.eu-north-1.compute.internal    <none>           <none>
pod/redis-74f4fb978c-2sb5t     1/1     Running     0          9h      192.168.3.209    ip-192-168-9-242.eu-north-1.compute.internal    <none>           <none>
pod/web-748fdd5fb9-46kpp       1/1     Running     0          7h52m   192.168.3.147    ip-192-168-9-242.eu-north-1.compute.internal    <none>           <none>
pod/worker-b4bf677dd-8qs6w     1/1     Running     0          67m     192.168.92.92    ip-192-168-86-95.eu-north-1.compute.internal    <none>           <none>
pod/worker-b4bf677dd-8wtzn     1/1     Running     0          67m     192.168.71.116   ip-192-168-86-95.eu-north-1.compute.internal    <none>           <none>
pod/worker-b4bf677dd-gm4vt     1/1     Running     0          67m     192.168.44.68    ip-192-168-33-205.eu-north-1.compute.internal   <none>           <none>
pod/worker-b4bf677dd-gvmz7     1/1     Running     0          67m     192.168.44.190   ip-192-168-33-205.eu-north-1.compute.internal   <none>           <none>
pod/worker-b4bf677dd-nbtzr     1/1     Running     0          43m     192.168.45.10    ip-192-168-33-205.eu-north-1.compute.internal   <none>           <none>
pod/worker-b4bf677dd-p2s6t     1/1     Running     0          145m    192.168.29.216   ip-192-168-9-242.eu-north-1.compute.internal    <none>           <none>
pod/worker-b4bf677dd-rlcwq     1/1     Running     0          67m     192.168.31.171   ip-192-168-9-242.eu-north-1.compute.internal    <none>           <none>
pod/worker-b4bf677dd-w79tq     1/1     Running     0          58m     192.168.89.131   ip-192-168-86-95.eu-north-1.compute.internal    <none>           <none>
celery                5.3.6
Django                4.2.13
django-celery-beat    2.6.0
django-celery-results 2.5.1
django-timezone-field 7.2.1
djangorestframework   3.15.1
psycopg2-binary       2.9.10
redis                 5.0.3


k8s/results/smoke/config_after.txt
NAME                       READY   UP-TO-DATE   AVAILABLE   AGE     CONTAINERS   IMAGES                                                      SELECTOR
deployment.apps/postgres   1/1     1            1           5h31m   postgres     postgres:12                                                 app=postgres
deployment.apps/redis      1/1     1            1           5h31m   redis        redis:7                                                     app=redis
deployment.apps/web        1/1     1            1           3h46m   web          <ACCOUNT_ID>.dkr.ecr.eu-north-1.amazonaws.com/jobqueue:v1   app=web
deployment.apps/worker     1/1     1            1           3h46m   worker       <ACCOUNT_ID>.dkr.ecr.eu-north-1.amazonaws.com/jobqueue:v1   app=worker

NAME                           READY   STATUS      RESTARTS   AGE     IP               NODE                                            NOMINATED NODE   READINESS GATES
pod/jobqueue-migrate-fxbxt     0/1     Completed   0          3h53m   192.168.46.226   ip-192-168-33-205.eu-north-1.compute.internal   <none>           <none>
pod/locust-smoke-284fz         1/1     Running     0          2m21s   192.168.44.68    ip-192-168-33-205.eu-north-1.compute.internal   <none>           <none>
pod/monitor                    1/1     Running     0          6m36s   192.168.46.226   ip-192-168-33-205.eu-north-1.compute.internal   <none>           <none>
pod/postgres-cf9fdbc59-tfwlk   1/1     Running     0          5h31m   192.168.85.124   ip-192-168-86-95.eu-north-1.compute.internal    <none>           <none>
pod/redis-74f4fb978c-2sb5t     1/1     Running     0          5h31m   192.168.3.209    ip-192-168-9-242.eu-north-1.compute.internal    <none>           <none>
pod/web-748fdd5fb9-46kpp       1/1     Running     0          3h46m   192.168.3.147    ip-192-168-9-242.eu-north-1.compute.internal    <none>           <none>
pod/worker-b4bf677dd-5hzfd     1/1     Running     0          10m     192.168.92.92    ip-192-168-86-95.eu-north-1.compute.internal    <none>           <none>


config/settings.py
from pathlib import Path
from decouple import config, Csv
from celery.schedules import crontab

# Build paths inside the project like this: BASE_DIR / 'subdir'.
BASE_DIR = Path(__file__).resolve().parent.parent


# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/4.2/howto/deployment/checklist/

# SECURITY WARNING: keep the secret key used in production secret!


SECRET_KEY = config('SECRET_KEY')
DEBUG = config('DEBUG', default = False, cast = bool)

# SECURITY WARNING: don't run with debug turned on in production!
# DEBUG = True

# ALLOWED_HOSTS = []
# ALLOWED_HOSTS = ['*']
ALLOWED_HOSTS = config('ALLOWED_HOSTS', default='localhost,127.0.0.1', cast=Csv())


# Application definition

INSTALLED_APPS = [
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    # Third-party
    'rest_framework',
    'django_celery_results',
    # Local
    'jobs',
]

MIDDLEWARE = [
    'django.middleware.security.SecurityMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
]

ROOT_URLCONF = 'config.urls'

TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'DIRS': [],
        'APP_DIRS': True,
        'OPTIONS': {
            'context_processors': [
                'django.template.context_processors.debug',
                'django.template.context_processors.request',
                'django.contrib.auth.context_processors.auth',
                'django.contrib.messages.context_processors.messages',
            ],
        },
    },
]

WSGI_APPLICATION = 'config.wsgi.application'


# Database
# https://docs.djangoproject.com/en/4.2/ref/settings/#databases

# DATABASES = {
#     'default': {
#         'ENGINE': 'django.db.backends.sqlite3',
#         'NAME': BASE_DIR / 'db.sqlite3',
#     }
# }

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.postgresql',
        'NAME': config('DB_NAME', default='jobqueue_db'),
        'USER': config('DB_USER', default='postgres'),
        'PASSWORD': config('DB_PASSWORD'),
        'HOST': config('DB_HOST', default='localhost'),
        'PORT': config('DB_PORT', default='5432'),
        # 'PORT': '6432',  # PgBouncer port, not 5432
        # 'CONN_MAX_AGE': 60, 
        # 'CONN_MAX_AGE': 0,  # As PgBouncer manage pooling, not Django

    }
}



# Redis & Celery
REDIS_URL = config('REDIS_URL', default='redis://localhost:6379/0')

CELERY_BROKER_URL = REDIS_URL
CELERY_RESULT_BACKEND = 'django-db'          # stores results in PostgreSQL
CELERY_CACHE_BACKEND = 'django-cache'
CELERY_ACCEPT_CONTENT = ['json']
CELERY_TASK_SERIALIZER = 'json'
CELERY_RESULT_SERIALIZER = 'json'
CELERY_TIMEZONE = 'Asia/Kolkata'
CELERY_TASK_TRACK_STARTED = True             # important for timing
CELERY_BROKER_CONNECTION_RETRY_ON_STARTUP = True

#Pririty queue cnfiguration
CELERY_TASK_QUEUES = {
    'jobs.high': {'exchange':'jobs', 'routing_keys':'high'},
    'jobs.default': {'exchange':'jobs', 'routing_keys':'default'},
    'jobs.low': {'exchange':'jobs', 'routing_keys':'low'},
}

CELERY_TASK_DEFAULT_QUEUE = 'jobs.default'

CELERY_TASK_ROUTES = {'jobs.tasks.process_job': {'queue': 'jobs.default'}}


CELERY_BEAT_SCHEDULE = {
    'cleanup-old-jobs': {
        'task': 'jobs.tasks.cleanup_old_jobs',
        'schedule': crontab(hour=2, minute=0),
    },
    'log-queue-stats': {
        'task': 'jobs.tasks.log_queue_stats',
        'schedule': 180.0,
    },
}
#to log those from the celery cron tasks
LOGGING = {
    'version': 1,
    'handlers': {
        'console': {
            'class': 'logging.StreamHandler',
        },
    },
    'root': {
        'handlers': ['console'],
        'level': 'INFO',
    },
}

CACHES = {
    'default' : {
        'BACKEND' : 'django.core.cache.backends.redis.RedisCache',
        'LOCATION' : REDIS_URL
    }
}
# Password validation
# https://docs.djangoproject.com/en/4.2/ref/settings/#auth-password-validators

# AUTH_PASSWORD_VALIDATORS = [
#     {
#         'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
#     },
#     {
#         'NAME': 'django.contrib.auth.password_validation.MinimumLengthValidator',
#     },
#     {
#         'NAME': 'django.contrib.auth.password_validation.CommonPasswordValidator',
#     },
#     {
#         'NAME': 'django.contrib.auth.password_validation.NumericPasswordValidator',
#     },
# ]

# DRF
REST_FRAMEWORK = {
    'DEFAULT_RENDERER_CLASSES': ['rest_framework.renderers.JSONRenderer'],
    'DEFAULT_PAGINATION_CLASS': 'rest_framework.pagination.PageNumberPagination',
    'PAGE_SIZE': 20,
}

# Internationalization
# https://docs.djangoproject.com/en/4.2/topics/i18n/

LANGUAGE_CODE = 'en-us'

TIME_ZONE = 'Asia/Kolkata'

USE_I18N = True

USE_TZ = True


# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/4.2/howto/static-files/

STATIC_URL = 'static/'

# Default primary key field type
# https://docs.djangoproject.com/en/4.2/ref/settings/#default-auto-field

DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'
